src/cart/crt_rpc.h

```diff
@@ -579,26 +579,23 @@ CRT_RPC_DECLARE(crt_ctl_log_add_msg, CRT_ISEQ_CTL_LOG_ADD_MSG,
 /* Internal macros for crt_req_(add|dec)ref from within cart.  These take
  * a crt_internal_rpc pointer and provide better logging than the public
  * functions however only work when a private pointer is held.
+ *
+ * Note that we conservatively use the default, strict memory order for both
+ * RPC_ADDREF and RPC_DECREF, leaving relaxations to future work.
  */
 #define RPC_ADDREF(RPC) do {						\
 		int __ref;						\
-		D_SPIN_LOCK(&(RPC)->crp_lock);				\
-		D_ASSERTF((RPC)->crp_refcount != 0,			\
-			  "%p addref from zero\n", (RPC));		\
-		__ref = ++(RPC)->crp_refcount;				\
-		D_SPIN_UNLOCK(&(RPC)->crp_lock);			\
-		RPC_TRACE(DB_NET, RPC, "addref to %d.\n", __ref);	\
+		__ref = atomic_fetch_add(&(RPC)->crp_refcount, 1);	\
```

Comment:
This would have been slightly better as memory_order_relaxed.

Great that we can new use standard atomics, we could probably remove gurt/atomic.h how althought we might need to keep the ATOMIC define https://github.com/daos-stack/daos/blob/master/src/include/gurt/atomic.h

Reply:
There's no synchronisation point here, the refcount is not used to pass data or unlock any other threads only to know when to free anything so "relaxed" should be correct.